go-controller/pkg/cni/cnishim.go

```diff
@@ -194,7 +199,120 @@ func (p *Plugin) CmdDel(args *skel.CmdArgs) error {
 }
 
 // CmdCheck is the callback for 'checking' container's networking is as expected.
-// Currently not implemented, so returns `nil`.
 func (p *Plugin) CmdCheck(args *skel.CmdArgs) error {
+	var err error
+
+	startTime := time.Now()
+	defer func() {
+		p.postMetrics(startTime, CNICheck, err)
+	}()
+
+	// read the config stdin args
+	conf, err := config.ReadCNIConfig(args.StdinData)
+	if err != nil {
+		return fmt.Errorf("invalid stdin args")
+	}
+	setupLogging(conf)
+	if gtet, err := version.GreaterThanOrEqualTo(conf.CNIVersion, "0.4.0"); err != nil {
+		klog.Errorf(err.Error())
+		return types.NewError(types.ErrDecodingFailure, err.Error(), "")
+	} else if !gtet {
+		err = types.NewError(types.ErrIncompatibleCNIVersion, "config version does not allow CHECK", "")
+		klog.Errorf(err.Error())
+		return err
+	}
+
+	// Parse previous result.
+	if conf.RawPrevResult == nil {
+		return fmt.Errorf("required prevResult missing")
+	}
+	if err := version.ParsePrevResult(&conf.NetConf); err != nil {
+		return err
+	}
+
+	netns, err := ns.GetNS(args.Netns)
+	if err != nil {
+		return fmt.Errorf("failed to open netns %q: %v", args.Netns, err)
+	}
+	defer netns.Close()
+
+	if conf.IPAM.Type != "" {
+		// run the IPAM plugin and get back the config to apply
+		err = ipam.ExecCheck(conf.IPAM.Type, args.StdinData)
+		if err != nil {
+			return err
+		}
+	}
+
+	result, err := current.NewResultFromResult(conf.PrevResult)
+	if err != nil {
+		return err
+	}
+
+	var contMap current.Interface
+	for _, intf := range result.Interfaces {
+		if args.IfName == intf.Name {
+			if args.Netns == intf.Sandbox {
+				contMap = *intf
+				break
+			}
+		}
+	}
+	// The namespace must be the same as what was configured
+	if args.Netns != contMap.Sandbox {
```

Comment:
I don't think this condition is possible.

Reply:
maybe it is impossible, but we can leave the test just in case, like in the plugin examples.